### DISCUSSION TITLE
CSV Extension?

### DISCUSSION THREAD

**Discussion by gomigg07** [TurboWarp/extensions#395] [Ideas]
An extension similar to the numerical encoding extension, but instead allows to read and write CSV strings.

An example of a practical use for this would be taking two lists, `particleEventsX` and `particleEventsY` and being able to combine them into one list, `particleEvents`, containing both x and y values in a single item on the list.

given a string `30|100|90|`, the `[read value]` block could be used like so:

`[set CSV to (string)]`
`[read value]` - reads first value 30
`[set x to (value)]`
`[read value]` - reads the next value 100
`[set y to (value)]`
`[read value]` - read final value 90
`[point in direction (value)]`

To be more flexible you could even make a `[write (value) with discriminator ( | )]` but it might be inconvenient.

PS: This is already possible in vanilla scratch with custom blocks, but I thought it would be convienient to shorten them down to an extension so you don't need to copy the `define` scripts for each sprite you want to use it in
![image](https://github.com/TurboWarp/extensions/assets/91437192/9d31c7f1-5274-480a-a533-b6fd52b5b916)

**Comment by ghost** [comment]
Maybe it's better to use JSON?
